Dockerfile

```diff
@@ -12,8 +13,6 @@ RUN npm install
 
 COPY . .
 
-RUN npm run build -- --catalogUrl=$catalogURL
-
 EXPOSE 8080
 
-CMD [ "serve", "-s", "dist", "-l", "8080" ]
\ No newline at end of file
+ENTRYPOINT [ "/bin/sh", "-c", "cd /app && npm run build -- --catalogUrl=${CATALOG_URL} --pathPrefix=${ROOT_PATH} && serve -s dist -l 8080" ]
```

Comment:
I'd prefer to inject `CATALOG_URL` and `ROOT_PATH` into the container as environment variables.  This is the more standard way of doing this.  @m-mohr is it possible to pass these as env variables to `npm run build` instead of CLI flags?

If environment variables are not an option I am fine with the proposed changes.

Reply:
STAC Browser will not recognize them automatically if you don't pass them via CLI parameters, but can't you use the env variables as CLI parameters?

Reply:
Thoughts on this @geospatial-jeff?

Reply:
This is still not very standard.  If we pass `CATALOG_URL` and `ROOT_PATH` as environment variables to the docker entrypoint we can run the docker container like this:

```
docker run -t stac-browser:latest -e CATALOG_URL -e ROOT_PATH
```

But if I ever want to override the docker command I have to either (1) not pass environment variables or (2) include environment variables in the command.  The problem is the usage of the container is conditional on whether or not you are overriding the command, and that is a big anti-pattern in docker.

I have concerns about building the app when the docker container is run instead of when the image is built.  Fast container start time is really important with docker, and this goes against the Vue documentation shown here https://v2.vuejs.org/v2/cookbook/dockerize-vuejs-app.html.

There are also inherent security risks with allowing the application to be built at runtime (ex. injection of malicious code).

Vue.js supports loading environment variables into the application at runtime (https://stackoverflow.com/questions/59722631/passing-environment-variables-at-runtime-to-vue-js-application-with-docker-compo).  My recommendation is to update the stac-browser code to accept `CATALOG_URL` and `ROOT_PATH` from either environment variables or through CLI flags and that would make this much easier to package in a container.  For example, these two commands would produce the same result:

```
# With CLI flags
npm run build -- --catalogUrl=catalogUrl

# With env variables
CATALOG_URL=catalogUrl npm run build --
```

But I'm not familiar with Vue and not sure if this is even possible.  For what its worth, it doesn't look like Vue (in general) has good support for distribution via docker and there are a ton of issues related to this online.

Reply:
I found this package which appears to solve the problem but in a really hacky way https://github.com/juanise/jvjr-docker-env#readme

Reply:
@geospatial-jeff Could you check whether this PR helps?
https://github.com/radiantearth/stac-browser/pull/329